### PR TITLE
dash/dg2-cell-editing-dbl-click

### PR DESCRIPTION
**Pull request by ddragula**
- [x] Refactored cell events.
- [x] Changed cell edit trigger to double click.
- Postponed: cell focusing (single click) - A11y & keyboard navigation reasons.
- [x] Added tests.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208280310748935

### CODE REVIEW

**Comment by ddragula** [comment]
Thanks for the comment! On the one hand, I want to prepare the basis for accessibility, on the other, I don't want to do too much so as not to introduce changes that will have to be rolled back due to bad implementation that breaks a11y more than it fixes - I am not an expert on this topic, but I think it's important to do it soon.

I would also like to make sure that some sort of navigation etc. is actually possible before we conduct external a11y testing. I'll contact @marvin19 about this task, and I'll leave just the double-click edit with the focus off 'until Wednesday' if I can't discuss navigation and a11y in more detail by then.

File: ts/DataGrid/Globals.ts [hunk 1]
```diff
@@ -76,6 +76,7 @@ namespace Globals {
         columnElement: classNamePrefix + 'column',
         hoveredCell: classNamePrefix + 'hovered-cell',
         hoveredColumn: classNamePrefix + 'hovered-column',
+        editedCell: classNamePrefix + 'edited-cell',
         focusedCell: classNamePrefix + 'focused-cell',
```
Comment (karolkolodziej) [inline ts/DataGrid/Globals.ts:80]: this can be removed

**Review by karolkolodziej — CHANGES_REQUESTED** [review]
Looks like a solid foundation for future a11y work! 
I get a few failing cypress tests locally- could you please check?

**Comment by ddragula** [comment]
I temporarily removed the features related to focusing because we want to merge this today (and several valuable changes have been made). This way, we avoid introducing an incomplete focus state without keyboard navigation. I'll move this to the next PR, which will go directly to the `master` branch if needed.

**Review by stitot — APPROVED** [review]
Looking good!

**Review by sebastianbochan — CHANGES_REQUESTED** [review]
DoubleClick is really tricky on touch devices (in many cases it does not work and some workarounds are needed).

Please test it on:
- native iPad (Safari + FF)
- desktop Firefox

**Review by karolkolodziej — APPROVED** [review]

**Comment by ddragula** [comment]
@sebastianbochan Tested, works well on FF, Safari, iPhone.

**Review by sebastianbochan — APPROVED** [review]

**Pull request merged**